index.js

```diff
@@ -63,15 +63,14 @@ function prebuildify (opts, cb) {
   // over the local .bin folder. Counter that by (again) adding .bin to PATH.
   opts.env = npmRunPath.env({ env: opts.env, cwd: opts.cwd })
 
-  mkdirp(opts.builds, function (err) {
-    if (err) return cb(err)
+  mkdirp(opts.builds).then(function () {
     loop(opts, function (err) {
       if (err) return cb(err)
 
       if (opts.artifacts) return copyRecursive(opts.artifacts, opts.builds, cb)
       return cb()
     })
-  })
+  }).catch(function (err) { return cb(err) })
```

Comment:
Let's instead use [`mkdirp-classic`](https://www.npmjs.com/package/mkdirp-classic)

Reply:
Why do you want to use the old callback version? Promises are much better.

Reply:
The deprecation simply means the author doesn't want to maintain it anymore. It still works just fine.

Reply:
> The deprecation simply means the author doesn't want to maintain it anymore

and we should upgrade to the higher version.

I do not have strong opinions here, but I do not understand the reason for not upgrading! 😄 I just want to know why the codebase of prebuild looks so legacy. Even the packages still use `var` instead of `let/consst`!

Reply:
Yep, let's use classic. The above use of promises has a re-entry bug with the cb being called twice, if something throws in the then

Reply:
@aminya re comment, everything works, everything is low maintenance, it's a cli, effort better spent elsewhere

Reply:
Fixed in https://github.com/prebuild/prebuildify/pull/45/commits/1d92d5ff0566a3395b99d687eccda38cd9554023

Reply:
> everything works, everything is low maintenance, it's a cli, effort better spent elsewhere

@mafintosh Things seem stable for others. However, my three efforts for using prebuild have resulted in three issues which then followed three PRs as my attempts for fixing them. 

https://github.com/prebuild/node-gyp-build/pull/35
https://github.com/prebuild/prebuildify/pull/45
https://github.com/prebuild/prebuildify/pull/46
https://github.com/prebuild/node-gyp-build/issues/22
https://github.com/prebuild/prebuildify/issues/10